### ISSUE TITLE
Exploding loss function

### ISSUE THREAD

**Issue by jpsember** [jpsember/java-ml#56]

**Comment by jpsember** [comment]
Gradient clipping?

https://androidkt.com/how-to-apply-gradient-clipping-in-pytorch/

**Comment by jpsember** [comment]
Why is the norm_giou reporting so many values?  Should be masked with 'obj_true'?
```
iou
[        │        │        │        │        │        │        │        │        │        │        │        │        ]
[        │        │        │        │        │        │        │        │        │        │        │        │        ]
[        │        │        │        │        │        │        │        │        │        │        │        │        ]
[        │        │        │  0.217 │        │        │        │        │        │        │        │        │        ]
[        │        │        │        │        │        │        │        │        │        │        │        │        ]
[        │        │        │        │        │        │        │        │        │        │        │        │        ]
[        │        │        │        │        │        │  0.499 │        │        │        │        │        │        ]
[        │        │        │        │        │        │        │        │        │        │        │        │        ]
[        │        │        │        │        │        │        │        │        │        │        │        │        ]
[        │        │        │        │        │        │        │        │        │        │        │        │        ]
[        │        │        │        │        │        │        │        │        │        │        │        │        ]
[        │        │        │        │        │        │        │        │        │        │        │        │        ]
[        │        │        │        │        │        │        │        │        │        │        │        │        ]
norm_giou
[  0.413 │  0.435 │  0.500 │  0.500 │  0.500 │  0.499 │  0.433 │  0.366 │  0.367 │  0.436 │  0.460 │  0.416 │  0.500 ]
[  0.500 │  0.472 │  0.413 │  0.500 │  0.418 │  0.496 │  0.415 │  0.486 │  0.377 │  0.500 │  0.445 │  0.488 │  0.424 ]
[  0.462 │  0.381 │  0.436 │  0.500 │  0.486 │  0.500 │  0.500 │  0.479 │  0.483 │  0.500 │  0.490 │  0.487 │  0.478 ]
[  0.500 │  0.401 │  0.419 │  0.538 │  0.482 │  0.386 │  0.500 │  0.485 │  0.473 │  0.500 │  0.410 │  0.500 │  0.461 ]
[  0.445 │  0.348 │  0.487 │  0.459 │  0.382 │  0.500 │  0.500 │  0.473 │  0.500 │  0.474 │  0.500 │  0.494 │  0.429 ]
[  0.490 │  0.450 │  0.447 │  0.425 │  0.500 │  0.492 │  0.466 │  0.456 │  0.496 │  0.434 │  0.457 │  0.482 │  0.472 ]
[  0.430 │  0.500 │  0.426 │  0.412 │  0.494 │  0.490 │  0.729 │  0.490 │  0.498 │  0.500 │  0.500 │  0.500 │  0.414 ]
[  0.459 │  0.360 │  0.482 │  0.424 │  0.458 │  0.471 │  0.500 │  0.416 │  0.500 │  0.500 │  0.493 │  0.500 │  0.500 ]
[  0.427 │  0.428 │  0.499 │  0.500 │  0.419 │  0.500 │  0.500 │  0.500 │  0.498 │  0.500 │  0.500 │  0.500 │  0.482 ]
[  0.500 │  0.464 │  0.500 │  0.412 │  0.388 │  0.473 │  0.500 │  0.472 │  0.426 │  0.474 │  0.500 │  0.349 │  0.485 ]
[  0.425 │  0.481 │  0.429 │  0.336 │  0.461 │  0.500 │  0.500 │  0.459 │  0.482 │  0.500 │  0.499 │  0.494 │  0.434 ]
[  0.451 │  0.381 │  0.500 │  0.471 │  0.385 │  0.359 │  0.397 │  0.457 │  0.500 │  0.458 │  0.485 │  0.423 │  0.362 ]
[  0.405 │  0.500 │  0.418 │  0.499 │  0.424 │  0.500 │  0.488 │  0.492 │  0.494 │  0.500 │  0.454 │  0.336 │  0.469 ]
```

**Comment by jpsember** [comment]
Was not logging the supplied tensor (the one that had the mask applied)
